### PR TITLE
Update task IAM role model to support Single Region Accounts

### PR DESCRIPTION
**Pull request by Yiyuanzzz**
### Summary

Update task IAM role model to support Single Region Accounts
### Implementation details

Update IAM role credentials model passed from ACS to the agent to include the `CredentialScope` field in `ecs-agent/credentials/manager.go` and existing tests.  The updated field will be passed through from the agent to the customer application container via the ecs-agent container credentials handler in `/ecs-agent/tmds/handlers/v1/credentials_handler.go`.  `CredentialScope` will be return as part of JSON response by credentials handler along with other credentials.

### Testing

Unit, integration, and functional tests.
New tests cover the changes:  no

### Description for the changelog

Update task IAM role model 

**Does this PR include breaking model changes? If so, Have you added transformation functions?**
  

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: ecs-agent/acs/session/session_test.go [hunk 1]
```diff
@@ -95,7 +95,8 @@ const (
           "expiration": "2016-03-25T06:17:19.318+0000",
           "roleArn": "r1",
           "secretAccessKey": "secretAccessKey",
-          "sessionToken": "token"
+          "sessionToken": "token",
+		  "credentialScope": "scope"
```
Comment (Realmonia) [inline ecs-agent/acs/session/session_test.go:99]: nit: indentation?

**Review by Realmonia — DISMISSED** [review]

File: ecs-agent/tmds/handlers/v1/credentials_handler.go [hunk 2]
```diff
@@ -144,8 +144,8 @@ func processCredentialsRequest(
 	credentialsJSON, err := json.Marshal(credentials.IAMRoleCredentials)
 	if err != nil {
 		errText := errPrefix + "Error marshaling credentials"
-		seelog.Errorf("Error processing credential request credentialType=%s taskARN=%s: %s",
-			credentials.IAMRoleCredentials.RoleType, credentials.ARN, errText)
+		seelog.Errorf("Error processing credential request credentialType=%s taskARN=%s credentialScope=%s: %s",
```
Comment (danehlim) [inline ecs-agent/tmds/handlers/v1/credentials_handler.go:147]: nit: I know using seelog is the existing behavior here, but typically we prefer to use the structured logger defined in `github.com/aws/amazon-ecs-agent/ecs-agent/logger` for logging in ecs-agent module. Could we use the structured logger for logging in this file to follow that convention and help make the logs more readable and traceable?

Example for reference: https://github.com/aws/amazon-ecs-agent/pull/3696/files#diff-b75f163bbc1723848662c8e77f544e15ebac5045354e2a4af8b057b502523252L386-R389

Comment (Yiyuanzzz) [reply ecs-agent/tmds/handlers/v1/credentials_handler.go:147]: Sure! I will update it

Comment (danehlim) [reply ecs-agent/tmds/handlers/v1/credentials_handler.go:147]: Synced offline, @Yiyuanzzz will switch to use structured logger in a followup pull request as this is non-blocking to this pull request.

File: ecs-agent/acs/session/session_test.go (hunk 1 above)
Comment (Yiyuanzzz) [reply ecs-agent/acs/session/session_test.go]: fixed, thanks

**Review by Realmonia — APPROVED** [review]

**Review by danehlim — APPROVED** [review]

**Pull request merged**